src/steps/prepareTags.ts

```diff
@@ -0,0 +1,40 @@
+import type { PluginConfigType } from './types';
+
+const validateCustomTagsSetting = (
+  customTags: unknown
+): string[] | undefined => {
+  if (!customTags) {
+    // No error message as the setting can just be simply not provided
+    return;
+  }
+
+  if (!Array.isArray(customTags)) {
+    throw new Error(
+      `customTags setting is not an array! customTags: ${customTags}`
+    );
+  }
+
+  if (!customTags.every((i) => typeof i === 'string')) {
+    throw new Error(
+      `customTags setting does not contain strings in array! customTags: ${customTags}`
+    );
+  }
+
+  return customTags;
+};
+
+const prepareTags = (version: string, pluginConfig: PluginConfigType) => {
+  const customTags = validateCustomTagsSetting(pluginConfig?.customTags);
+  const tagsFormat = customTags || ['v${major}'];
+
+  const [major, minor, patch] = version.split('.');
+
+  return tagsFormat.map((tag) =>
+    tag
+      .replace('${major}', major)
+      .replace('${minor}', minor)
+      .replace('${patch}', patch)
```

Comment:
Would have expected [lodash templating here](https://lodash.com/docs/4.17.15#template) similar to how semantic-release uses it.

This might cause unexpected behavior with the following strings:

- `'v${ major }'` (expected `'v1'`)
- `'v${major}-beta-pre${major}'` (expected `'v1-beta-pre1'`)

Reply:
Thanks for this 💪  
I initially followed this example and didn't want to overcomplicate it:
https://semantic-release.gitbook.io/semantic-release/usage/configuration#tagformat 
But your examples are perfectly good points as to supporting both `${ x }` and `${x}-${x}`. If I have some extra time some time soon I'll add support for both :D If you'd like to add it then feel free to open a PR and I'll review ASAP, but don't worry about it if you don't have much time :D  
Cheers